diff --git a/src/device.c b/src/device.c
index ad446b2..04261eb 100644
--- a/src/device.c
+++ b/src/device.c
@@ -983,7 +983,12 @@ uvc_error_t uvc_claim_if(uvc_device_handle_t *devh, int idx) {
 
   /* Tell libusb to detach any active kernel drivers. libusb will keep track of whether
    * it found a kernel driver for this interface. */
+#if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
+  /* Keep the audio interfaces attached on iOS. */
+  ret = LIBUSB_ERROR_NOT_FOUND;
+#else
   ret = libusb_detach_kernel_driver(devh->usb_devh, idx);
+#endif
 
   if (ret == UVC_SUCCESS || ret == LIBUSB_ERROR_NOT_FOUND || ret == LIBUSB_ERROR_NOT_SUPPORTED) {
     UVC_DEBUG("claiming interface %d", idx);
@@ -1026,7 +1031,12 @@ uvc_error_t uvc_release_if(uvc_device_handle_t *devh, int idx) {
   if (UVC_SUCCESS == ret) {
     devh->claimed &= ~( 1 << idx );
     /* Reattach any kernel drivers that were disabled when we claimed this interface */
+#if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
+    /* The iOS path does not detach a driver. */
+    ret = UVC_SUCCESS;
+#else
     ret = libusb_attach_kernel_driver(devh->usb_devh, idx);
+#endif
 
     if (ret == UVC_SUCCESS) {
       UVC_DEBUG("reattached kernel driver to interface %d", idx);
